net/dns/manager.go

```diff
@@ -265,16 +265,23 @@ func (m *Manager) compileConfig(cfg Config) (rcfg resolver.Config, ocfg OSConfig
 	rcfg.Routes = routes
 	ocfg.Nameservers = []netip.Addr{cfg.serviceIP()}
 
-	if m.os.SupportsSplitDNS() {
-		ocfg.MatchDomains = cfg.matchDomains()
-	} else {
+	var bcfg *OSConfig
+	isApple := runtime.GOOS == "darwin" || runtime.GOOS == "ios"
+	if isApple || !m.os.SupportsSplitDNS() {
 		// If the OS can't do native split-dns, read out the underlying
 		// resolver config and blend it into our config.
-		bcfg, err := m.os.GetBaseConfig()
-		if err != nil {
+		g, err := m.os.GetBaseConfig()
+		if err == nil {
+			bcfg = &g
+		} else if !isApple || err != ErrGetBaseConfigNotSupported {
```

Comment:
is this right? It's at best confusing to read.

Maybe an extra if statement for just `else if isApple` with nothing in it would be easier to read?

Reply:
Yeah I was just trying to rewrite that as well. How's this?